presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ControllerConfigAPI.java

```diff
@@ -0,0 +1,181 @@
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<Integer> nodeId)
+    {
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConfigAPI.class)
+                .pathMethod("getConfigFile")
+                .httpMethod(GET)
+                .resolveTemplate("file", file)
+                .accept(MediaType.TEXT_PLAIN)
+                .build();
+
+        return getResponseFromRequest(scope, apiRequester, nodeId);
+    }
+
+    @GET
+    @Path("/{file}/{property}")
+    @Produces({MediaType.TEXT_PLAIN})
+    public Response getConfigProperty(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            @PathParam("property") @ApiParam("A specific property") String property,
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<Integer> nodeId)
+    {
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConfigAPI.class)
+                .pathMethod("getConfigProperty")
+                .httpMethod(GET)
+                .resolveTemplate("file", file)
+                .resolveTemplate("property", property)
+                .accept(MediaType.TEXT_PLAIN)
+                .build();
+
+        return getResponseFromRequest(scope, apiRequester, nodeId);
+    }
+
+    @POST
+    @Path("/{file}")
+    @Produces({MediaType.TEXT_PLAIN})
+    public Response setConfigFileByURL(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            String url,
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<Integer> nodeId)
+    {
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConfigAPI.class)
+                .pathMethod("setConfigFileByURL")
+                .httpMethod(POST)
+                .resolveTemplate("file", file)
+                .accept(MediaType.TEXT_PLAIN)
+                .entity(Entity.entity(url, MediaType.TEXT_PLAIN))
+                .build();
+
+        return getResponseFromRequest(scope, apiRequester, nodeId);
+    }
+
+    @DELETE
+    @Path("/{file}")
+    @Produces({MediaType.TEXT_PLAIN})
+    public Response deleteConfigFile(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            @QueryParam("scope") String scope,
+            @QueryParam("nodeId") List<Integer> nodeId)
+    {
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConfigAPI.class)
+                .pathMethod("deleteConfigFile")
+                .httpMethod(DELETE)
+                .resolveTemplate("file", file)
+                .accept(MediaType.TEXT_PLAIN)
+                .build();
+
+        return getResponseFromRequest(scope, apiRequester, nodeId);
+    }
+
+    private Response getResponseFromRequest(String scope, ApiRequester apiRequester, Collection<Integer> nodeId)
+    {
+        if (((scope != null) && (!nodeId.isEmpty())) || (scope == null && nodeId.isEmpty())) {
```

Comment:
We should decide whether the default behavior should be to target the entire cluster, or whether a scope should be required for every request.

In the former case, the checks after the `||` here can be removed.